packages/clerk-js/bundle-check.mjs

```diff
@@ -0,0 +1,238 @@
+//@ts-check
+import fs from 'node:fs';
+import http from 'node:http';
+import path from 'node:path';
+import { pipeline } from 'node:stream';
+import zlib from 'node:zlib';
+
+import { chromium } from 'playwright';
+
+/**
+ * This script generates a CLI report detailing the gzipped size of JavaScript resources loaded by `clerk-js` for a
+ * given configuration. This is useful to ensure that the total amount of loaded JavaScript does not exceed the
+ * anticipated amount for a particular invocation.
+ */
+
+/**
+ * The base HTML file that each case will be executed against. Loads `clerk-js` from the local server in addition to
+ * creating an instance of `VirtualRouter`.
+ * @param {string} script The case-specific code to execute, typically to mount a component like `SignIn`.
+ */
+function template(script) {
+  return `
+<!DOCTYPE html>
+<html>
+    <body>
+        <div id="app"></div>
+        <script
+          type="text/javascript"
+          src="/clerk.browser.js"
+          data-clerk-publishable-key="${process.env.CLERK_PUBLISHABLE_KEY}"
+        ></script>
+        <script type="text/javascript">
+        class VirtualRouter {
+            name = 'VirtualRouter';
+            mode = 'virtual';
+
+            #url;
+            #listeners = new Set();
+
+            constructor(path) {
+                const origin = typeof window === 'undefined' ? 'https://clerk.dummy' : window.location.origin;
+
+                this.#url = new URL(path ?? '/', origin);
+            }
+
+            push(path) {
+                const newUrl = new URL(this.#url.toString());
+                newUrl.pathname = path;
+
+                this.#url = newUrl;
+                this.emit();
+            }
+
+            replace(path) {
+                this.push(path);
+            }
+
+            shallowPush(path) {
+                this.push(path);
+            }
+
+            pathname() {
+                return this.#url.pathname;
+            }
+
+            searchParams() {
+                return this.#url.searchParams;
+            }
+
+            subscribe(listener) {
+                this.#listeners.add(listener);
+
+                return () => this.#listeners.delete(listener);
+            }
+
+            emit() {
+                this.#listeners.forEach(listener => listener(this.#url));
+            }
+
+            getSnapshot() {
+                return this.#url;
+            }
+        }
+        window.VIRTUAL_ROUTER = new VirtualRouter('/');
+        </script>
+        <script type="text/javascript">${script}</script>
+    </body>
+</html>
+`;
+}
+
+function signIn() {
+  const script = `
+window.Clerk.load({ router: window.VIRTUAL_ROUTER }).then(() => {
```

Comment:
This `router` option is not actually used in this version of `clerk-js`. It's being introduced in #4114 but it doesn't hurt anything to have it here now. Happy to remove it if it gives anyone heartburn!